File: app/src/main/java/org/myspecialway/ui/splash/SplashActivity.java
package org.myspecialway.ui.splash;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import org.myspecialway.App;
import org.myspecialway.ui.main.MainScreenActivity;
import org.myspecialway.ui.login.LoginActivity;
import org.myspecialway.ui.login.RequestCallback;
import org.myspecialway.session.UserSession;

public class SplashActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
//        Notifications.createNotificationChannel(this);

        App.Companion.getInstance().getUserSessionManager().refreshSessionIfNeeded(new RequestCallback<UserSession>() {

            @Override
            public void onSuccess(UserSession result) {
                startMainActivity();
            }

            @Override
            public void onFailure(Throwable t) {
                startLoginActivity();
            }
        });
    }

    private void startMainActivity() {
        Intent intent = new Intent(this, MainScreenActivity.class);
        startActivity(intent);
        finish();
    }

    private void startLoginActivity(){
        Intent intent = new Intent(this, LoginActivity.class);
        startActivity(intent);
        finish();
    }
}

File: app/src/main/java/org/myspecialway/ui/splash/SplashActivity.kt
package org.myspecialway.ui.splash

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import org.koin.android.ext.android.inject
import org.myspecialway.App
import org.myspecialway.common.Navigation
import org.myspecialway.session.SessionManager
import org.myspecialway.session.UserSession
import org.myspecialway.ui.login.LoginRepository
import org.myspecialway.ui.login.gateway.RequestCallback

class SplashActivity : AppCompatActivity() {

    private val sessionManager: SessionManager by inject()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        when(sessionManager.isLoggedIn) {
            true -> Navigation.toMainActivity(this@SplashActivity)
            false -> Navigation.toLoginActivity(this@SplashActivity)
        }
    }
}
